action.yaml

```diff
@@ -162,6 +162,15 @@ runs:
         path: ${{ github.workspace }}/.autobuild-installables
         key: ${{ runner.os }}-${{ runner.arch }}-${{ inputs.configuration }}-${{ hashFiles('autobuild.xml') }}
 
+    - name: Determine branch
+      id: which-branch
+      shell: bash
+      run: |
+        # in real use, 'git branch -r --contains' should produce a single line,
+        # but our self-tests can emit more than one
+        branch="$(git branch -r --contains ${{ steps.sha.outputs.long }} | head -n 1)"
```

Comment:
is there an advantage of this git incantation over `git rev-parse --abbrev-ref HEAD` which has been the standard one that I've included in my shell prompt since time immemorial?

Reply:
Likely to report detached HEAD. `git branch --contains <hash>` reports detached HEAD unless you also supply `-r`.